test/revisions_test.go

```diff
@@ -0,0 +1,105 @@
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/arangodb/go-driver"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestRevisionTree(t *testing.T) {
+	if getTestMode() != testModeSingle {
+		t.Skipf("Not a single")
+	}
+	c := createClientFromEnv(t, true)
+	skipBelowVersion(c, "3.7", t)
+
+	db := ensureDatabase(nil, c, "revision_tree", nil, t)
+	col := ensureCollection(nil, db, "revision_tree", nil, t)
+
+	var noOfDocuments int = 80000
+	expectedDocuments := make([]interface{}, 0, noOfDocuments)
+	for i := 0; i < noOfDocuments; i++ {
+		expectedDocuments = append(expectedDocuments, UserDoc{
+			Name: fmt.Sprintf("User%d", 1),
+			Age:  i,
+		})
+	}
+
+	_, _, err := col.CreateDocuments(context.Background(), expectedDocuments)
+	require.NoErrorf(t, err, "Failed to create new documents: %s", describe(err))
+
+	batch, err := c.Replication().CreateBatch(context.Background(), db, 123, time.Hour)
+	require.NoError(t, err)
+	defer batch.Delete(context.Background())
+
+	getTree := func() (driver.RevisionTree, error) {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+
+		return c.Replication().GetRevisionTree(timeoutCtx, db, batch.BatchID(), col.Name())
+	}
+
+	tree, err := getTree()
+	if err != nil {
+		if err.Error() == "this collection doesn't support revision-based replication" {
```

Comment:
Here it would probably be better to compare the error number, since messages can occasionally be changed slightly, also, it is more efficient.

Reply:
changed